integrationtests/self/http_test.go

```diff
@@ -957,3 +957,118 @@ func TestHTTPStreamer(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, PRData, repl)
 }
+
+func TestHTTPClientTrace(t *testing.T) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/client-trace", func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(100 * time.Millisecond)
+		w.WriteHeader(http.StatusContinue)
+	})
+	port := startHTTPServer(t, mux)
+
+	buf := make([]byte, 1)
+	getConn := false
+	gotConn := false
+	gotFirstResponseByte := false
+	got100Continue := false
+	got1xxResponse := false
+	dnsStart := false
+	dnsDone := false
+	connectStart := false
+	connectDone := false
+	tlsHandshakeStart := false
+	tlsHandshakeDone := false
+	wroteHeaderField := false
+	wroteHeaders := false
+	wait100Continue := false
+	wroteRequest := false
+	trace := httptrace.ClientTrace{
+		GetConn: func(hostPort string) {
+			getConn = true
+			require.Equal(t, hostPort, fmt.Sprintf("localhost:%d", port))
```

Comment:
It's invalid to use `testing.T` concurrently.

Reply:
fixed